src/components/Vote.jsx

```diff
@@ -4,29 +4,31 @@ import { updateArticleVotes } from '../Utils/patchApi';
 const Vote = ({ article_id, initialVotes }) => {
   const [votes, setVotes] = useState(initialVotes);
   const [voteError, setVoteError] = useState(null);
+  const [hasVoted, setHasVoted] = useState(false);
 
-  const handleVote = (voteType) => {
+  const handleVote = () => {
     let inc_votes;
 
-    if (voteType === 'upvote') {
-      inc_votes = 1;
-    } else if (voteType === 'downvote') {
+    if (hasVoted) {
       inc_votes = -1;
+    } else {
+      inc_votes = 1; 
     }
 
     
     setVotes(votes + inc_votes);
+    setHasVoted(!hasVoted); 
 
     updateArticleVotes(article_id, inc_votes)
       .then(() => {
         
-        setVotes(votes + inc_votes);
         setVoteError(null); 
       })
       .catch((error) => {
        
-        setVotes(votes);
-        setVoteError('Failed to update votes. Please try again.'); 
+        setVotes(votes - inc_votes);
```

Comment:
Try using a callback here instead of votes - inc_votes - this will stop it from decreasing by 2 when it reverts.

Reply:
see the setSentMessageCount here: https://notes.northcoders.com/courses/js-front-end/ui-behaviours-and-optimistic-rendering#optimistic-approach